src/test/java/org/takes/rs/RsPrintTest.java

```diff
@@ -72,30 +73,77 @@ public void flushBodyEvenWhenExceptionHappens() throws IOException {
     /**
      * RsPrint can flush head contents even when exception happens.
      * @throws IOException If some problem inside
-     * @todo #778:30min Fix flushHeadEvenWhenExceptionHappens test
-     *  and remove Ignore annotation. Implementation could not rely
-     *  on simple try-finally block as OutputStreamWriter do not call
-     *  to the flush() of OutputStream. It has additional checks.
      */
     @Test
-    @Ignore
     public void flushHeadEvenWhenExceptionHappens() throws IOException {
         final IOException exception = new IOException("Error");
-        final FailOutputStream output = new FailOutputStream(exception);
+        final FailWriter writer = new FailWriter(exception);
```

Comment:
@victornoel why can't we keep this test as is? Changing the test may miss some potential failures, which were covered with previous test.